ls1intum/Artemis
# Enable / Disable second correction with button

### Checklist

### Motivation and Context
Enable instructions to dynamically start / stop the second correction

### Description
With this PR, we implement a button for exams with two correction rounds, which allows instructors to enable / disable the possibility for tutors to assess within the second correction round. 

### Steps for Testing


Only for instructors: 
1. Log in to Artemis
2. Navigate to Course Administration -> Course -> Exam -> an Exam with second correction 
3. Second correction button should be shown in the two views
4. Clicking the button enables / disables the second correction row. It activates / deactivates the view of the second correction, it does *not* delete existing results and more. 
5. You can activate second correction, review results within the second correction. When deactivating and reactivating it, the existing results are shown as usual
6. Navigate to Course Administration -> Course -> Exam -> an Exam without second correction 
7. No second correction button should be shown in the two views

For tutors: 
If activated by an instructor, the second correction is possible. If not activated, its not shown / possible. 
### Test Coverage

exercise-assessment-dashboard: 80.2% 
second-correction-enable-button.component.ts: 100%
ExerciseResource: 95%

### Screenshots

![Bildschirmfoto 2021-01-28 um 17 24 24](https://user-images.githubusercontent.com/42205689/106167654-b632d600-618d-11eb-9d7a-52668c00508b.png)
![Bildschirmfoto 2021-01-28 um 17 15 11](https://user-images.githubusercontent.com/42205689/106167666-b92dc680-618d-11eb-85be-fbd19ed4294d.png)
